code/game/machinery/shieldgen.dm

```diff
@@ -553,3 +553,644 @@
+
+	var/turf/connected_turf
+
+/obj/machinery/modular_shield/module/Initialize(mapload)
+	. = ..()
+
+	connected_turf = get_step(loc, dir)
+
+/obj/machinery/modular_shield/module/Destroy()
+	. = ..()
+	if(shield_generator)
+		shield_generator.connected_modules -= (src)
+		shield_generator.calculate_boost()
+	if(connected_node)
+		connected_node.connected_through_us -= (src)
+	return ..()
+
+/obj/machinery/modular_shield/module/attackby(obj/item/I, mob/user, params)
+
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		panel_open = !(panel_open)
+		I.play_tool_sound(src, 50)
+		update_icon_state()
+		if(panel_open)
+			to_chat(user, span_notice("You open the maintenance hatch of [src]."))
+			return TRUE
+		to_chat(user, span_notice("You close the maintenance hatch of [src]."))
+		return TRUE
+
+//rather than automatically checking for connections its probably alot less expensive to just make the players manually multi tool sync each part
+	if(I.tool_behaviour == TOOL_MULTITOOL)
+		try_connect(user)
+		return
+
+	if(default_change_direction_wrench(user, I))
+		if(shield_generator)
+			shield_generator.connected_modules -= (src)
+			shield_generator.calculate_boost()
+			shield_generator = null
+			update_icon_state()
+		if(connected_node)
+			connected_node.connected_through_us -= (src)
+			connected_node = null
+		connected_turf = get_step(loc, dir)
+		return
+
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/modular_shield/module/setDir(new_dir)
+	. = ..()
+	connected_turf = get_step(loc, dir)
+
+/obj/machinery/modular_shield/module/proc/try_connect(user)
+
+	if(shield_generator || connected_node)
+		balloon_alert(user, "already connected to something")
+		return
+
+	shield_generator = (locate(/obj/machinery/modular_shield_gen) in connected_turf)
+
+	if(shield_generator)
+
+		shield_generator.connected_modules += (src)
+		balloon_alert(user, "connected directly to generator")
+		update_icon_state()
+		if(istype(src, /obj/machinery/modular_shield/module/node))
+			var/obj/machinery/modular_shield/module/node/connected_node = src
+			connected_node.connect_connected_through_us()
+		shield_generator.calculate_boost()
+		return
+
+	connected_node	= (locate(/obj/machinery/modular_shield/module/node) in connected_turf)
+
+	if(connected_node)
+
+		connected_node.connected_through_us += (src)
+		shield_generator = connected_node.shield_generator
+		if(shield_generator)
+			shield_generator.connected_modules += (src)
+			balloon_alert(user, "connected to generator through node")
+			update_icon_state()
+			if(istype(src, /obj/machinery/modular_shield/module/node))
+				var/obj/machinery/modular_shield/module/node/connected_node = src
+				connected_node.connect_connected_through_us()
+			shield_generator.calculate_boost()
+			return
+		balloon_alert(user, "connected to node but no path to generator")
+		return
+	balloon_alert(user, "failed to detect powered generator or node")
```

Comment:
these balloon alerts are way too long, make them more succinct or change it to a `to_chat`.